### ISSUE TITLE
Automatic Forecasting and Exogenous Features

### ISSUE THREAD

**Issue by SinghJivjot** [Nixtla/neuralforecast#845]
### Description

The current Hyperparameter Optimization is excellent. One suggestion would be to allow us to pass the Exogenous Features separately to an AutoModel like AutoGRU, as one would do while fitting the Model like GRU, but keeping in mind if I want to use the default Grid-Params (thru Ray or Optuna) I can still do that and pass the Exogenous Features independently.

### Use case

It would look something like this I guess

from neuralforecast.models import GRU
from neuralforecast.auto import AutoGRU
from neuralforecast.losses.pytorch import DistributionLoss

GRU(
    h=30,
    loss=DistributionLoss(),
    futr_exog_list = [],
    hist_exog_list = [],
    stat_exog_list = [],
)

AutoGRU(
    h=30,
    loss=DistributionLoss(),
    config = None,
    futr_exog_list = [],
    hist_exog_list = [],
    stat_exog_list = [],
)

Here I can keep config parameter to None, thereby using the default HPO grid set by you guys and pass in my extra features without touching the config parameter.
PS: This is my first time raising a Issue on GitHub. Apologies If i didn't adhere to any rules/protocol

**Comment by jmoralez** [comment]
Hey @SinghJivjot, thanks for using neuralforecast. The exog lists are in the config so that they can be tuned as well, if you want to use the default config and override the exog lists you could do something like this:
```python
from neuralforecast.auto import AutoGRU
from neuralforecast.losses.pytorch import DistributionLoss

exogs = {'futr_exog_list': [], 'hist_exog_list': [], 'stat_exog_list': []}
model = AutoGRU(h=30, loss=DistributionLoss('Poisson'))
model.config.update(exogs)
```

**Comment by SinghJivjot** [comment]
Oh, silly me trying to over think such simple work arounds. Thanks a lot
